website/www/site/content/en/documentation/io/built-in/google-bigquery.md

```diff
@@ -602,59 +602,108 @@ as the previous example.
+[different data ingestion options](https://cloud.google.com/bigquery/loading-data)
+(specifically, [load jobs](https://cloud.google.com/bigquery/docs/batch-loading-data)
+and [streaming inserts](https://cloud.google.com/bigquery/streaming-data-into-bigquery))
 for more information about these tradeoffs.
 
+{{< paragraph class="language-java" >}}
 BigQueryIO chooses a default insertion method based on the input `PCollection`.
+You can use `withMethod` to specify the desired insertion method. See
+[`Write.Method`](https://beam.apache.org/releases/javadoc/{{< param release_latest >}}/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.Write.Method.html)
+for the list of the available methods and their restrictions.
+{{< /paragraph >}}
 
 {{< paragraph class="language-py" >}}
-BigQueryIO uses load jobs when you apply a BigQueryIO write transform to a
-bounded `PCollection`.
+BigQueryIO chooses a default insertion method based on the input `PCollection`.
+You can use `method` to specify the desired insertion method. See
+[`WriteToBigQuery`](https://beam.apache.org/releases/pydoc/{{< param release_latest >}}/apache_beam.io.gcp.bigquery.html#apache_beam.io.gcp.bigquery.WriteToBigQuery)
+for the list of the available methods and their restrictions.
 {{< /paragraph >}}
 
-{{< paragraph class="language-java" >}}
 BigQueryIO uses load jobs in the following situations:
-{{< /paragraph >}}
 
 {{< paragraph class="language-java" wrap="span" >}}
 * When you apply a BigQueryIO write transform to a bounded `PCollection`.
-* When you apply a BigQueryIO write transform to an unbounded `PCollection` and
-  use `BigQueryIO.write().withTriggeringFrequency()` to set the triggering
-  frequency.
 * When you specify load jobs as the insertion method using
   `BigQueryIO.write().withMethod(FILE_LOADS)`.
 {{< /paragraph >}}
 
-{{< paragraph class="language-py" >}}
-BigQueryIO uses streaming inserts when you apply a BigQueryIO write transform to
-an unbounded `PCollection`.
+{{< paragraph class="language-py" wrap="span" >}}
+* When you apply a BigQueryIO write transform to a bounded `PCollection`.
+* When you specify load jobs as the insertion method using
+  `WriteToBigQuery(method='FILE_LOADS')`.
 {{< /paragraph >}}
 
+***Note:*** If you use batch loads in a streaming pipeline:
+
 {{< paragraph class="language-java" >}}
-BigQueryIO uses streaming inserts in the following situations:
+You must use `withTriggeringFrequency` to specify a triggering frequency for
+initiating load jobs. Be careful about setting the frequency such that your
+pipeline doesn't exceed the BigQuery load job [quota limit](https://cloud.google.com/bigquery/quotas#load_jobs).
+{{< /paragraph >}}
+
+{{< paragraph class="language-java" >}}
+You can either use `withNumFileShards` to explicitly set the number of file
+shards written, or use `withAutoSharding` to enable dynamic sharding (starting
+2.29.0 release) and the number of shards may be determined and changed at
```

Comment:
Does it make sense to put the release version here?

Reply:
yes, this is fine : )